### PR TITLE
core.get_objects_inside_radius: Omit removed objects

### PR DESCRIPTION
**Pull request by HybridDog**
See #6294

### CODE REVIEW

File: doc/lua_api.txt [hunk 1]
```diff
@@ -2469,8 +2469,9 @@ and `minetest.auth_reload` call the authetification handler.
 * `minetest.add_item(pos, item)`: Spawn item
     * Returns `ObjectRef`, or `nil` if failed
 * `minetest.get_player_by_name(name)`: Get an `ObjectRef` to a player
-* `minetest.get_objects_inside_radius(pos, radius)`
+* `minetest.get_objects_inside_radius(pos, radius)`: returns a list of object references
     * `radius`: using an euclidean metric
+    * This function skips removed objects.
```
Comment (SmallJoker) [inline doc/lua_api.txt]: Redundant IMO. We know the background story and thus, why this should be documented. But will assume this behaviour anyway, as they expect items to be removed instantly.

File: doc/lua_api.txt [hunk 2]
```diff
@@ -2469,8 +2469,9 @@ and `minetest.auth_reload` call the authetification handler.
 * `minetest.add_item(pos, item)`: Spawn item
     * Returns `ObjectRef`, or `nil` if failed
 * `minetest.get_player_by_name(name)`: Get an `ObjectRef` to a player
-* `minetest.get_objects_inside_radius(pos, radius)`
+* `minetest.get_objects_inside_radius(pos, radius)`: returns a list of object references
```
Comment (SmallJoker) [inline doc/lua_api.txt]: `object references` -> \``ObjectRef`\`

Comment (DS-Minetest) [reply doc/lua_api.txt]: ``` `ObjectRef` ``` is singular, so ``` `ObjectRef`s ``` (or ``` `ObjectRefs` ```, I don't know).

File: doc/lua_api.txt (hunk 1 above)
Comment (HybridDog) [reply doc/lua_api.txt]: See obj:remove(), it says that items are removed when returning from lua. So when modders use this function, call objects' remove and then call this function again with the same parameters, they know that the remove()d objects are not returned.

File: doc/lua_api.txt (hunk 2 above)
Comment (HybridDog) [reply doc/lua_api.txt]: ok

**Pull request merged**